### PR TITLE
[Hub Generated] Review request for Microsoft.AlertsManagement to add version stable/2019-03-01

### PR DESCRIPTION
**Pull request by orieldar**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/orieldar/azure-rest-api-specs/dev-alertsmanagement-Microsoft.AlertsManagement-2019-03-01...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [x] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by orieldar** [comment]
Hi @xccc-msft ,
Thanks in advance for the review.

I know the changes are considered a breaking change, however these are bug fixes for a misalignment between the ARM Manifest and the Alerts Management swagger.
smartGroups isn’t exposed in our manifest under 2019-03-01 so the swagger should be fixed to not expose them under that api-version.  That should not be considered a breaking change as any clients trying to use that api-version would fail.

**Comment by xccc-msft** [comment]
@orieldar Could you please help update [CODEOWNERS](https://github.com/Azure/azure-rest-api-specs/blob/master/CODEOWNERS) for AlertsManagement?

**Comment by orieldar** [comment]
> @orieldar Could you please help update [CODEOWNERS](https://github.com/Azure/azure-rest-api-specs/blob/master/CODEOWNERS) for AlertsManagement?



> @orieldar Could you please help update [CODEOWNERS](https://github.com/Azure/azure-rest-api-specs/blob/master/CODEOWNERS) for AlertsManagement?

Sure, done.

**Comment by xccc-msft** [comment]
@orieldar Thanks. I put on the label of `BreakingChangeReviewRequired` as it's removing APIs in the PR. You may discuss more once the reviewers start to review.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json
Comment (EranGon) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json:329]: >subscriptions [](start = 6, length = 13)

just making sure - in our ARM manifest we also support two more resources for this API version:
smartDetectorAlertRules
alertsMetaData

Are they defined in another file?

**Review by EranGon — APPROVED** [review]
:shipit:

**Comment by orieldar** [comment]
all tests pass except breakingchange, need your approval @iscai-msft @assing @xccc-msft, can you please review?

**Review by xseeseesee — APPROVED** [review]

**Comment by xccc-msft** [comment]
@orieldar We still need ARM review. I'm from SDK team.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json
Comment (pilor) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json:55]: The response defined in this spec does not conform to the RPC.  This is a collection GET and should follow the "paging response body" schema here:  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json
Comment (pilor) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json:1247]: Each "alertMetadata" resource should look like a resource with a name, type, and ID.  They should also have individual GETs if you are modeling them as a GETable resource.  Basically, make the API and definitions look a lot more like /alerts

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json
Comment (EranGon) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json:1247]: @pilor this change is only documenting the current API that our RP exposes. Would it be OK to approve this change, and we'll open a bug to fix the API in the coming sprints?

Comment (pilor) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2019-03-01/AlertsManagement.json:1247]: Any new api-versions for this API need to conform to the RPC.  You won't be able to make changes here in the future without implementing that

**Review by pilor — APPROVED** [review]
API somehow got released in november without API review... approving the current version to avoid breaking.  Future versions need to incorporate fixes

**Comment by xccc-msft** [comment]
@akning-ms Can you please check the context and help merge? Thanks.

**Comment by akning-ms** [comment]
> Hi @xccc-msft ,
> Thanks in advance for the review.
> 
> I know the changes are considered a breaking change, however these are bug fixes for a misalignment between the ARM Manifest and the Alerts Management swagger.
> smartGroups isn’t exposed in our manifest under 2019-03-01 so the swagger should be fixed to not expose them under that api-version. That should not be considered a breaking change as any clients trying to use that api-version would fail.

Just a check, can we know why smartGroups not exposed in manifest? if it is a mistake, can we fix it in manifest, instead of change swagger?

**Comment by EranGon** [comment]
> > Hi @xccc-msft ,
> > Thanks in advance for the review.
> > I know the changes are considered a breaking change, however these are bug fixes for a misalignment between the ARM Manifest and the Alerts Management swagger.
> > smartGroups isn’t exposed in our manifest under 2019-03-01 so the swagger should be fixed to not expose them under that api-version. That should not be considered a breaking change as any clients trying to use that api-version would fail.
> 
> Just a check, can we know why smartGroups not exposed in manifest? if it is a mistake, can we fix it in manifest, instead of change swagger?

@akning-ms the smartGroups resource is still in public preview, and as such should not be part of a stable version swagger. So this is intended

**Pull request merged**
